doc/api/assert.md

```diff
@@ -772,7 +772,7 @@ argument, then `error` is assumed to be omitted and the string will be used for
 <!-- eslint-disable no-restricted-syntax -->
 ```js
 // THIS IS A MISTAKE! DO NOT DO THIS!
-assert.throws(myFunction, 'missing foo', 'did not throw with expected message');
+assert.throws(myFunction, 'missing foo'); // Did not throw with expected message
```

Comment:
The text above this example needs to be re-written now as well as the example on line 778 as well. Rather than moving the third argument to a comment, perhaps just remove it entirely. The user should still be cautioned against using a string for the expected error message because a string will be interpreted as the message to provide if the function doesn't throw rather than something to validate the thrown error.

Reply:
I am not sure I can follow. The situation was only improved in case three arguments are passed to `throws`. Otherwise the problem continues to exist. So the mitigation is only very tiny as the typical wrong use case is probably `assert.throws(fn, "error msg")`.

I thought a few times about how to improve the two args situation but as far as I see it there is nothing that can be done to improve it. I actually also thought about deprecating using a string as second argument but that would be very inconsistent and weird as well.

Reply:
Refining my comment a bit: The current version of the example code (`assert.throws(fn, str, str)`) is obviously an error. The version above in this PR (`assert.throws(fn, str)`) is less obviously an error because maybe "missing foo" is not an expected error message but rather the error message we want to provide if the function doesn't throw as expected. It's not clear to me if the best way to clarify is to edit the text preceding the block or if it is to edit `missing foo` to say something else.